binaries/upload.py

```diff
@@ -63,29 +73,42 @@ def upload_conda_packages():
                     print(f"Anaconda package upload failed for pacakge {name}")
                     return exit_code
 
-    print(f"All packages uploaded to anaconda successfully under the channel {CONDA_USER}")
+    print(
+        f"All packages uploaded to anaconda successfully under the channel {CONDA_USER}"
+    )
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(
         description="Upload anaconda and pypi packages for torchserve and torch-model-archiver"
     )
     parser.add_argument(
-        "--upload-conda-packages", action="store_true", required=False, help="Specify whether to upload conda packages"
+        "--upload-conda-packages",
+        action="store_true",
+        required=False,
+        help="Specify whether to upload conda packages",
+    )
+    parser.add_argument(
+        "--upload-pypi-packages",
+        action="store_true",
+        required=False,
+        help="Specify whether to upload pypi packages",
     )
     parser.add_argument(
-        "--upload-pypi-packages", action="store_true", required=False, help="Specify whether to upload pypi packages"
+        "--test-pypi",
+        action="store_true",
+        required=False,
+        help="Specify whether to upload to test PyPI",
     )
     args = parser.parse_args()
 
     if args.upload_conda_packages:
         upload_conda_packages()
 
     if args.upload_pypi_packages:
-        upload_pypi_packages()
-    
+        upload_pypi_packages(args.test_pypi)
+
     if any([args.upload_conda_packages, args.upload_pypi_packages]):
```

Comment:
This check is strange, let's say the upload failed but didn't return the right error code then the print statement would say it was? Let's push these checks to the function doing the upload

I believe the `No packages uploaded` check intent was to have some `dry_run` command. I used this for Docker which you can also use https://github.com/pytorch/serve/blob/master/docker/docker_nightly.py#L11 - I also see some `-x` in the bash commands but again can I see this script running locally. Feel free to use what you think is best.

Reply:
So, currently if the upload to pypi fails , the script shows a failure. I am running this "set -ex ;" along with the twine upload command